### PR TITLE
[DOCS] Update docstring for date2num (module function)

### LINKED ISSUE #202: Wrong dtype of return in date2num
## 📚 Documentation

The documentation for [`cf_units.Unit.date2num`](https://github.com/SciTools/cf-units/blob/a69d7c81cc8a0da4dada191a0861020bd56f33f0/cf_units/__init__.py#L1944) and [`cf_units.date2num`](https://github.com/SciTools/cf-units/blob/a69d7c81cc8a0da4dada191a0861020bd56f33f0/cf_units/__init__.py#L352) stipulates that both functions return a `float, or numpy.ndarray of floats`. However, since the return value comes directly from `cftime.date2num`, this will regularly return "an array of integers" since cftime version 1.4.0 (see [the changelog](https://github.com/Unidata/cftime/blob/b3e5c740106051e7dcff500ee6de979091f5a37c/Changelog#L47)).

### PR DESCRIPTION
**Pull request by ukmo-ccbunney**
## 🚀 Pull Request
As per #467, but for module function `date2num`

The underlying cftime package will now preferentially return an integer rather than a float if if encoded datetime(s) can be expressed exactly as an integer.

### Description
Docstring for `cf_units.date2num` updated to reflect this underlying change in behaviour.
Extra example added showing outputs of type integer where encoded allows.

#467 fixed docstring for `cf_units.Unit.date2num`, but this module level function was missed.

Fixes: https://github.com/SciTools/cf-units/issues/202

### CODE REVIEW

File: cf_units/__init__.py [hunk 1]
```diff
@@ -365,6 +369,13 @@ def date2num(date, unit, calendar):
         ...               cf_units.CALENDAR_STANDARD)
         array([6.5, 7.5])
 
+        # Integer type preferentially returned if possible:
```
Comment (pp-mo) [inline cf_units/__init__.py]:
```suggestion
        >>> # Integer type preferentially returned if possible:
```

Now I got an html build to work, I find that this doesn't render correclty
(and neither does the original in #467 !)

It seems not to like the blank line + stops rendering as a code example.
If you prefix with the correctly indented ">>> ", that seems to fix it.

If you don't like that and would prefer a blank line, please do, but I'm not myself sure how to do that.

**Review by pp-mo — CHANGES_REQUESTED** [review]
Can you please fix the Unit.date2num similarly ?

It seems increasingly that I made a mess of reviewing #467 !

File: cf_units/__init__.py (hunk 1 above)
Comment (pp-mo) [reply cf_units/__init__.py]: Oh, on a roll here !!
I think you can use "\<BLANKLINE>" somehow : see [here](https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html#directive-doctest)

Comment (ukmo-ccbunney) [reply cf_units/__init__.py]: Ah yes - I see now that this is doctest expecting the comment line to be an output of the previous test.
Originally, I was running `make doctest` to get Sphinx to run the doctests and this was passing (even though it shouldn't which makes me think it wasn't __really__ running the doctests). 😲

Now I am running the doctests with `pytest --doctest-modules` which correctly catches this error.

I believe that `<BLANKLINE>` is only valid when you **expect** a blank line as a test output, so I think the best solution here is to simply ensure the comment and any blank lines are prepended with  `>>>`. However, for styling consistency, I have removed the blank line (but kept the comment)

I've done this for both the `cftime.date2num` and `cftime.Unit.date2num`

**Review by pp-mo — APPROVED** [review]
:+1: thanks @ukmo-ccbunney

**Comment by trexfeathers** [comment]
I have a fix for those pre-commit errors in #484, if you wanted to merge that one first?

**Comment by pp-mo** [comment]
( merge-in fixed the pre-commit failure. -- thanks @trexfeathers for #484 )

**Pull request merged**